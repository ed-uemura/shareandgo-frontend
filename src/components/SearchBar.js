import React, { useState, useContext } from 'react';
import { Context } from '../context/AuthContext';
import '../css/SearchBar.css';
import ButtonPrimary from './ButtonPrimary';

const SearchBar = ({ link }) => {
	const [values, setValues] = useState('');
	const { setDestination } = useContext(Context);

	function handleChange(event) {
		const auxValues = { ...values };
		auxValues[event.target.id] = event.target.value;
		setValues(auxValues);
	}

	function handleDestination() {
		setDestination(values.searchDestination);
	}

	return (
		<div className='search-destination col-12 col-md-8 mb-5 mx-auto'>
			<form>
				<div>
					<input
						type='text'
						id='searchDestination'
						placeholder='Where are you going?'
						className='form-control search-input'
						onChange={(e) => handleChange(e)}
					/>

					<ButtonPrimary
						text={<span className='material-symbols-outlined'>search</span>}
						className='btn-search'
						link={link}
						clickAction={handleDestination}
					/>
				</div>
			</form>
		</div>
	);
};

export default SearchBar;
