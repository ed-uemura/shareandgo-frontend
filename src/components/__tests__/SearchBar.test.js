import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import SearchBar from '../SearchBar'
import { MemoryRouter } from 'react-router-dom';


afterEach(() =>{
    cleanup();
});

test('matches snapshot', () =>{ 
    const tree = renderer.create(
        <MemoryRouter>
            <SearchBar />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});