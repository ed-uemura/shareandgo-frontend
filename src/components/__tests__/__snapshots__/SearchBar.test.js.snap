// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="search-destination col-12 col-md-8 mb-5 mx-auto"
>
  <form>
    <div>
      <input
        className="form-control search-input"
        id="search-destination"
        onChange={[Function]}
        placeholder="Where are you going?"
        type="text"
      />
      <a
        className="btn btn-primary btn-search"
        href="/passenger-destination"
        onClick={[Function]}
      >
        <span
          className="material-symbols-outlined"
        >
          search
        </span>
      </a>
    </div>
  </form>
</div>
`;
